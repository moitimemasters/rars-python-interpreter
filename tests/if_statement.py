a = 0
if 1 > 2:
    a += 0
elif 1 >= 2:
    a += 2
elif 1 <= 2:
    a += 3
else:
    a += 100

if a > 0:
    a += 100

if a == 0:
    a -= 100
else:
    a += 100

a
