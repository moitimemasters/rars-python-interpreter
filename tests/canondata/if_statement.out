file: tests/if_statement.py
started lexing
ended lexing
started parsing ast
address of if condition: 268766744
address of if condition: 268783000
address of if condition: 268788120
ended parsing ast
started compiling
ended compiling
compiled instructions:
	 Instruction: 2
	 Instruction: 8
	 Instruction: 2
	 Instruction: 2
	 Instruction: 7
	 Instruction: 6
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 8
	 Instruction: 5
	 Instruction: 2
	 Instruction: 2
	 Instruction: 7
	 Instruction: 6
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 8
	 Instruction: 5
	 Instruction: 2
	 Instruction: 2
	 Instruction: 7
	 Instruction: 6
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 8
	 Instruction: 5
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 8
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 6
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 8
	 Instruction: 5
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 6
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 8
	 Instruction: 5
	 Instruction: 0
	 Instruction: 2
	 Instruction: 7
	 Instruction: 8
	 Instruction: 0
started interpreting
ended interpreting
Interpretation result: 203
